docs/guides/using-the-network-shim.md

```diff
@@ -0,0 +1,193 @@
+# Using the Network Shim
+
+## About
+
+The Network Shim is a tool that uses the a cypress plugin and the cypress `cy.intercept()` API to intercept network traffic during a test run. It has three different _modes_:
+
+-   `capture`: when operating in this mode, the Network Shim will intercept all outgoing requests to the specified hosts and allow each request to reach the destination server. Each response is stored as a fixture (JSON file).
+-   `stub`: in this mode the Network Shim also intercepts all outgoing requests to the specified hosts, but it will prevent the request from reaching the destination server and instead it will produce a response from a previously recorded fixture.
+-   `live`: the Network Shim does not intercept any requests when this mode is enabled.
+
+By first recording fixtures during a _capture run_ and then using these fixtures during a _stub run_ it effectively becomes possible to execute a full cypress test suite without a live backend. This is especially relevant for running a cypress test suite on CI.
+
+A capture or stub run is executed for a specific DHIS2 api version and individual fixtures for each version are produced/used during a run. This paves the way for doing cross-version testing for _feature-toggling_ apps.
+
+## General workflow
+
+As mentioned, he Network Shim is a tool that enables a workflow where a cypress e2e test suite is executed on CI.
+
+The general steps required to setup a workflow like that are as follows:
+
+1. Ensure a live backend is up and running
+1. Ensure a fully passing cypress test-suite is in place
+1. Ensure the GitHub workflow includes a job that runs the cypress test suite on CI in stub mode
+1. Execute a capture run, which will produce fixture files
+1. Commit the generated fixture files and push to the remote
+1. The test suite will now execute on CI and if a change breaks a test it will be impossible to merge
+
+To setup cross-version testing essentially means taking some of the steps above multiple times:
+
+-   Perform multiple capture runs against different versions of the live backend, so network fixtures are created for all these versions
+-   Define multiple jobs in the GitHub workflow that execute a stub run against these different versions
+
+## Configuration
+
+A workflow as described above consists of a few moving parts, which are described in more details below. **However, running the `d2-utils-cypress install` command should create all required files**.
+
+### Cypress
+
+#### `./cypress.env.json`
+
+The network shim expects a few environment variables to be present. These can be set using the `--env` flag when interacting with the cypress CLI, but base values should be provided as well the `./cypress.env.json` configuration file:
+
+```json
+{
+    "dhis2Username": "YOUR_USERNAME_HERE",
+    "dhis2Password": "YOUR PASSWORD HERE",
+    "dhis2BaseUrl": "http://localhost:8080",
```

Comment:
`dhis2BaseUrl` is required by the NetworkShim too, but you are right about the other properties. Personally I think it's OK to keep them in, since I already mention that in order to use the Network Shim you will also need the `enableAutoLogin` setup and the `chromeAllowXSiteCookies` plugin.

Reply:
Resolving this, as it was discussed on slack